src/cpx_genericcallbacks.jl

```
@@ -0,0 +1,260 @@
+"""
+function cbsetfunc(model::Model, context_mask::Clong, callback_func::Function)
+    c_callback = @cfunction(
+        callback_wrapper, Cint, (Ptr{Cvoid}, Clong, Ptr{Cvoid})
+    )
+    user_handle = (model, callback_func)::Tuple{Model, Function}
+    ret = @cpx_ccall(callbacksetfunc,
+        Cint,
+        (Ptr{Cvoid}, Ptr{Cvoid}, Clong, Ptr{Cvoid}, Any),
+        model.env, model.lp, context_mask, c_callback, user_handle
+    )
+    if ret != 0
+        throw(CplexError(model.env, ret))
+    end
+    # We need to keep a reference to the callback function so that it isn't
+    # garbage collected.
+    model.callback = user_handle
+    return ret
+end
+
+function cbabort(callback_data::CallbackContext)
+    return_status = @cpx_ccall(callbackabort,
+                    Cint,
+                    (Ptr{Cvoid},),
+                    callback_data.context)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetrelaxationpoint(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int, obj::Cdouble)
+    return_status = @cpx_ccall(callbackgetrelaxationpoint,
+                Cint,
+                (Ptr{Cvoid}, Ptr{Cdouble}, Cint, Cint, Ref{Cdouble}),
+                callback_data.context, x, begin_index -= 1, end_index -= 1, obj)
+    if return_status != 0
+        throw(CplexError(callback_data.model.env, return_status))
+    end
+    return return_status
+end
+
+function cbpostheursoln(callback_data::CallbackContext, cnt::Int, ind::Vector{Int}, val::Vector{Cdouble}, obj::Cdouble, strat::CbSolStrat)
+    return_status = @cpx_ccall(callbackpostheursoln,
+                    Cint,
+                    (Ptr{Cvoid}, Cint, Ptr{Cint}, Ptr{Cdouble}, Cdouble, Cint),
+                    callback_data.context, cnt, ind .-= 1, val, obj, strat)
+    # The following codes should be added back when the error CPLEX 1003 is resolved.
```

Can you elaborate? We certainly shouldn't just return `Cint(0)`.

---

CPLEX error 1003 is a bad argument error, which indicates that an invalid argument is passed. However, I rechecked all the argument and did not see any bad argument. Plus, despite the error code, the function is able to post the correct heuristic solution. 

So I was wondering if we could just ignore this error code for now, and get back to it if someone reports an issue on this.

---

Please put back the error handling for now and we'll see how it goes.